core-rust/mesh-api-server/src/mesh_api/handlers/construction_parse.rs

```diff
@@ -48,8 +54,12 @@ pub(crate) async fn handle_construction_parse(
     };
 
     let mapping_context = MappingContext::new(&state.network);
-    let database = state.state_manager.database.snapshot();
-    let operations = to_mesh_api_operations_from_instructions_v1(&instructions, &mapping_context, database.deref())?;
+    let database = state.state_manager.database.access_direct();
```

Comment:
Is this safe?

i.e. do we access the state database? If yes, we should use snapshot. If not, we should comment explaining why this is okay.

Reply:
Decided to revert to `snapshot()` as per [this conversation](https://rdxworks.slack.com/archives/C071C6DB8TG/p1733841098517859?thread_ts=1732877188.486819&cid=C071C6DB8TG)